docs/rhel4edge_iso.md

```diff
@@ -202,6 +204,123 @@ oc get cs
+
+The default ISO image is configured to run a Caddy HTTP server for the `ostree`
+updates at the `http://127.0.0.1:8080` URL, serving the contents of the local
+`/var/lib/ostree-server-local` directory containing an empty `ostree` update
+repository.
+
+Log into the MicroShift server using `redhat:redhat` credentials and run the
+following commands to check the local `ostree` server configuration.
+
+```bash
+$ systemctl is-active ostree-server-local.service
+active
+
+$ sudo journalctl -u ostree-server-local.service --output cat
+Started Caddy HTTP server for the local ostree repository.
+...
+...
+
+$ ostree remote list
+edge
+
+$ ostree remote show-url edge
+http://127.0.0.1:8080/repo
+
+$ ostree remote summary edge
+Repository Mode (ostree.summary.mode): archive-z2
+Last-Modified (ostree.summary.last-modified): 2023-04-28T04:55:59-04
+Has Tombstone Commits (ostree.summary.tombstone-commits): No
+ostree.summary.indexed-deltas: true
+```
+
+> The default `ostree` local repository is empty and it does not contain any
+> commit revisions to be installed.
+
+**Custom Server**
+
+This default behavior can be overriden by specifying the `-ostree_server_url`
+command line argument when running the `scripts/image-builder/build.sh` script.
+The URL parameter of this argument should point to a custom `ostree` server to
+be used for installing updates on an existing image.
+
+Log into the MicroShift server using `redhat:redhat` credentials and run the
+following commands to check the custom `ostree` server configuration and
+install updates.
+
+```bash
+$ ostree remote list
+edge
+
+$ ostree remote show-url edge
+<YOUR_OSTREE_SERVER_URL>
+
+$ ostree remote summary edge
+<YOUR_OSTREE_REPO_SUMMARY_AND_COMMIT_REV>
+
+$ sudo rpm-ostree deploy <YOUR_OSTREE_COMMIT_REV>
+...
+...
+# Reboot the system for the update to become active
+```
+
+**Local Updates**
+
+When the default Caddy HTTP server is configured for the `ostree` updates, users
+can overwrite the contents of the `/var/lib/ostree-server-local` directory and
+install the updates locally.
+
+One of the techniques for generating `ostree` updates is supported by the
+`scripts/image-builder/build.sh` script. When the `-build_edge_commit` command
+line argument is specified, the script builds an edge commit archive instead of
+an ISO image.
+
+```bash
+~/microshift/scripts/image-builder/build.sh -pull_secret_file ~/.pull-secret.json -build_edge_commit
+...
+...
+# Edge commit created
+The contents of the archive can be used for serving ostree updates:
+/home/microshift/microshift/_output/image-builder/microshift-container-0.0.1-commit.tar
+
+# Done
+```
+
+> Optionally specify additional `-microshift_rpms`, `-custom_rpms`, or
+> `-embed_containers` arguments to customize the edge commit archive contents.
+
+Copy the produced `microshift-container-0.0.1-commit.tar` archive to the
+MicroShift server and unpack it at the `/var/lib/ostree-server-local` directory,
+deleting the previous contents.
```

Comment:
How does a user stage multiple deployments this way? Does having rpm-ostree download the deployment leave it staged for rollback, or manually switching to/from it later?

Reply:
Currently, there is no option to have more than one update in the same commit. 
The rollback works unrelated to the ostree server because it's stored locally.

Reply:
Having mutliple commits is possible to arrange by adding another ostree source. Currently, we have the default `edge`, so we can add more and have all those available. 
Let me see if it's easy to document it.

Reply:
Actually, I did not find an option to merge more than one commit in the same repo.

Reply:
I think we may want to merge this in the current form and get feedback from users about possible other scenarios to support in the devenv. WDYT?